src/GitVersionCore/Extensions/GitExtensions.cs

```diff
@@ -99,27 +41,19 @@ public static void DumpGraph(string workingDirectory, Action<string> writer = nu
             }
         }
 
-        public static bool IsBranch(this string branchName, string branchNameToCompareAgainst)
+        public static bool IsBranch(this string branchName, string branchNameToCompare)
```

Comment:
`IsBranch` seems to belong in the `Branch` class, but also seems to do much of the same as `IsSameBranch`, only on strings instead. Is it often we only have `string` instances of branch names available and not the full `Branch` object we can use for comparison?

Reply:
Changed to finding the Branch by name then use Equals